networkit/cpp/io/FastSNAPGraphReader.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * FastSNAPGraphReader.cpp
+ *
+ *  Created on: 04.05.2018
+ *      Author: Alexander van der Grinten
+ */
+
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "FastSNAPGraphReader.h"
+#include "../auxiliary/StringTools.h"
+#include "../auxiliary/Log.h"
+
+namespace NetworKit {
+
+FastSNAPGraphReader::FastSNAPGraphReader(const bool& directed, const count& nodeCount, const bool& remapNodes) :
+directed(directed), nodeCount(nodeCount), remapNodes(remapNodes){}
+
+Graph FastSNAPGraphReader::read(const std::string &path) {
+	Graph graph(0,false,directed);
+
+	//In the actual state this parameter has very little influence on the reader performance.
+	//There can be a significant boost if it is possible to reserve space in the graph initialization
+	if(nodeCount != 0)
+		nodeIdMap.reserve(nodeCount);
```

Comment:
Shouldn't memory only be reserved when actually needed, i.e., when `remapNodes` is true, too?

Reply:
True, I fixed it in https://github.com/kit-parco/networkit/pull/168/commits/f944ddb670fdac2161c583d90acf6a2e0414604d